packages/quill/src/modules/clipboard.ts

```diff
@@ -26,10 +26,26 @@ import normalizeExternalHTML from './normalizeExternalHTML/index.js';
 
 const debug = logger('quill:clipboard');
 
-type Selector = string | Node['TEXT_NODE'] | Node['ELEMENT_NODE'];
-type Matcher = (node: Node, delta: Delta, scroll: ScrollBlot) => Delta;
+type TextSelector = Node['TEXT_NODE'];
+type ElementSelector = string | Node['ELEMENT_NODE'];
+type NodeSelector = string;
+type Selector = TextSelector | ElementSelector;
+
+type NodeMatcher = (node: Node, delta: Delta, scroll: ScrollBlot) => Delta;
+type TextMatcher = (node: Text, delta: Delta, scroll: ScrollBlot) => Delta;
+type ElementMatcher = (
+  node: Element,
+  delta: Delta,
+  scroll: ScrollBlot,
+) => Delta;
+type Matcher = TextMatcher | ElementMatcher | NodeMatcher;
 
-const CLIPBOARD_CONFIG: [Selector, Matcher][] = [
+type MatcherConfig =
+  | [TextSelector, TextMatcher]
+  | [ElementSelector, ElementMatcher]
```

Comment:
Not sure if it's safe in `ElementMatcher` to consider the first parameter as Element instead of Node because TextSelector and ElementSelector both contains `string` as the part of the union.

Reply:
@luin Let me know if you're still unsure about this change or whether you'd like time to consider it.  I'm happy to close the PR if you think it's better in its current state.